packages/world/src/systems/SpawnSystem.sol

```diff
@@ -16,16 +15,24 @@ import { Energy, EnergyData } from "../codegen/tables/Energy.sol";
 import { Mass } from "../codegen/tables/Mass.sol";
 import { ActionType } from "../codegen/common.sol";
 
-import { WORLD_DIM_X, WORLD_DIM_Z, SPAWN_ENERGY } from "../Constants.sol";
-import { AirObjectID, PlayerObjectID } from "../ObjectTypeIds.sol";
+import { WORLD_DIM_X, WORLD_DIM_Z, SPAWN_ENERGY, SPAWN_AREA_HALF_WIDTH } from "../Constants.sol";
+import { AirObjectID, PlayerObjectID, SpawnTileObjectID } from "../ObjectTypeIds.sol";
 import { checkWorldStatus, getUniqueEntity, gravityApplies, inWorldBorder } from "../Utils.sol";
 import { transferAllInventoryEntities } from "../utils/InventoryUtils.sol";
 import { notify, SpawnNotifData } from "../utils/NotifUtils.sol";
 import { getForceField } from "../utils/ForceFieldUtils.sol";
 import { TerrainLib } from "./libraries/TerrainLib.sol";
+import { callChipOrRevert } from "../utils/callChip.sol";
+import { updateMachineEnergyLevel } from "../utils/MachineUtils.sol";
+import { ISpawnTileChip } from "../prototypes/ISpawnTileChip.sol";
+
+import { VoxelCoord, VoxelCoordLib } from "../VoxelCoord.sol";
+
 import { EntityId } from "../EntityId.sol";
 
 contract SpawnSystem is System {
+  using VoxelCoordLib for *;
```

Comment:
hmm why is this needed if its already global?

Reply:
it is just for the PositionData.toVoxelCoord()